hummingbot/connector/exchange/bybit/bybit_api_user_stream_data_source.py

```diff
@@ -94,28 +91,128 @@ async def listen_for_user_stream(self, output: asyncio.Queue):
+            raise
+        except Exception:
+            self.logger().error(
+                "Unexpected error occurred subscribing to private channels...",
+                exc_info=True
+            )
+            raise
+
     async def _authenticate_connection(self, ws: WSAssistant):
         """
         Sends the authentication message.
         :param ws: the websocket assistant used to connect to the exchange
         """
-        auth_message: WSJSONRequest = WSJSONRequest(payload=self._auth.generate_ws_authentication_message())
-        await ws.send(auth_message)
+        request: WSJSONRequest = WSJSONRequest(
+            payload=self._auth.generate_ws_auth_message()
+        )
+        await ws.send(request)
 
     async def _process_ws_messages(self, ws: WSAssistant, output: asyncio.Queue):
         async for ws_response in ws.iter_messages():
             data = ws_response.data
-            if isinstance(data, list):
-                for message in data:
-                    if message["e"] in ["executionReport", "outboundAccountInfo"]:
-                        output.put_nowait(message)
-            elif data.get("auth") == "fail":
-                raise IOError("Private channel authentication failed.")
+            if "op" in data:
+                if data.get("op") == "auth":
+                    await self._process_ws_auth_msg(data, output)
+                elif data.get("op") == "pong":
+                    await self._process_ws_pingpong_msg(data, output)
+                elif data.get("op") == "subscribe":
+                    if data.get("success") is False:
+                        self.logger().error(
+                            "Unexpected error occurred subscribing to private channels...",
+                            exc_info=True
+                        )
+                continue
+            topic = data.get("topic")
+            channel = ""
+            if topic == CONSTANTS.WS_SUBSCRIPTION_ORDERS_ENDPOINT_NAME:
+                channel = CONSTANTS.PRIVATE_ORDER_CHANNEL
+            elif topic == CONSTANTS.WS_SUBSCRIPTION_EXECUTIONS_ENDPOINT_NAME:
+                channel = CONSTANTS.PRIVATE_TRADE_CHANNEL
+            elif topic == CONSTANTS.WS_SUBSCRIPTION_WALLET_ENDPOINT_NAME:
+                channel = CONSTANTS.PRIVATE_WALLET_CHANNEL
+            else:
+                pass
+            if channel:
+                data["channel"] = channel
+                output.put_nowait(data)
+
+    async def _process_ws_auth_msg(self, data: dict, output: asyncio.Queue):
+        # {
+        #     "success": true,
+        #     "ret_msg": "",
+        #     "op": "auth",
+        #     "conn_id": "XXX"
+        # }
+        if data.get("success") is False:
+            raise IOError("Private channel authentication failed.")
+        else:
+            self.logger().info("Private channel authentication success.")
+
+    async def _process_ws_pingpong_msg(self, data: dict, output: asyncio.Queue):
+        # {
```

Comment:
it's there is no need to process this one?

Reply:
Left unimplemented by mistake. Will push a commit for processing ping-pong messages for errors.